### PR TITLE
[IGNORE] Add doc about secret

### PR DESCRIPTION
**Pull request by Nexucis**
I'm starting a documentation about the `Secret` and the `GlobalSecret`.

### CODE REVIEW

File: docs/secret.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Secret
+
+When you are defining a datasource, likely you will need to provide a basic authentication, a certificate, or a token
```
Comment (AntoineThebaud) [inline docs/secret.md:3]:
```suggestion
When defining a datasource, you will probably need to provide a basic authentication, a certificate, or a token
```

File: docs/secret.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+to be used when the Perses backend will contact your Datasource.
+
+`Secret` or `GlobalSecret` are the two objects that you need. The choice between a `Secret` or
+a `GlobalSecret` will mainly depend on if you are defining a `Datasource` or a `GlobalDatasource`.
```
Comment (AntoineThebaud) [inline docs/secret.md:7]: Maybe we should highlight more the difference in terms of "scope behavior" with the rest of the resources here?
I mean e.g a project variable can rely on a global datasource for instance, while here we have:
- GlobalDatasource -> GlobalSecret
- (Project) Datasource -> Secret only
- Dashboard datasource -> didnt check but Secret only too I guess ?

Maybe it's an issue to have such discrepancy btw, not sure but anyway that's out of the topic of this PR.

WDYT?

Comment (Nexucis) [reply docs/secret.md:7]: ah yeah indeed it's certainly unclear like that. I will change that

File: docs/secret.md [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
-When you are defining a datasource, likely you will need to provide a basic authentication, a certificate, or a token
-to be used when the Perses backend will contact your Datasource.
+When defining a datasource, you will probably need to provide a basic authentication, a certificate, or a token to be
+used when the Perses backend will contact your Datasource.
 
-`Secret` or `GlobalSecret` are the two objects that you need. The choice between a `Secret` or
-a `GlobalSecret` will mainly depend on if you are defining a `Datasource` or a `GlobalDatasource`.
+We have two different objects to store sensitive data: `GlobalSecret` and `Secret`.
+To know what which one you need, it will depend on what object representing a datasource you are using.
```
Comment (AntoineThebaud) [inline docs/secret.md:7]:
```suggestion
You should use one or the other depending on which object your datasource corresponds to.
```

File: docs/secret.md [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
+
+If you want to store sensitive data for a `GlobalDatasource`, then you need to create a `GlobalSecret`.
```
Comment (AntoineThebaud) [inline docs/secret.md:9]:
```suggestion
- To store sensitive data for a `GlobalDatasource`, you need to create a `GlobalSecret`.
- For a `Datasource` object or a datasource defined directly in a dashboard , you need to create a `Secret`. `GlobalSecret` cannot be used here.
```

**Review by AntoineThebaud — APPROVED** [review]

**Pull request merged**